sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -175,6 +182,44 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
    * Override to provide additional rules for the operator optimization batch.
    */
   def extendedOperatorOptimizationRules: Seq[Rule[LogicalPlan]] = Nil
+
+  override def batches: Seq[Batch] = {
+    val excludedRules =
+      SQLConf.get.optimizerExcludedRules.toSeq.flatMap(_.split(",").map(_.trim).filter(!_.isEmpty))
```

Comment:
nit: `!_.isEmpty` -> `_.nonEmpty`

Reply:
Also, you need to handle case-sensitivity.

Reply:
There is an auto-generated field `ruleName` in `Rule`, so we do exact name matching (case sensitive).

Reply:
You can use `Utils.stringToSeq`?